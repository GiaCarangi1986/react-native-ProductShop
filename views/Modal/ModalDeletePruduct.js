import React from "react";
import { Button } from "react-native-elements";
import { Text } from "react-native";
import Wrapper from "../Wrappers";
import { MODAL_CONSTS } from "../../const";
import style from "./style";

const ModalDeletePruduct = ({ setModalVisible = () => { }, modalVisible = false, deleteProducts = () => { }, actionCancel = () => { } }) => {
  return (
    <>
      <Text style={style.text}>{MODAL_CONSTS.delete_product.title}</Text>
      <Wrapper nameOfStyle='btn_group'>
        <Button
          type='outline'
          buttonStyle={style.btn_delete}
          titleStyle={style.btn_delete_text}
          title={MODAL_CONSTS.delete_product.answer[0]}
          onPress={() => {
            deleteProducts()
            setModalVisible(!modalVisible)
          }}
        />
        <Button
          type='outline'
          buttonStyle={style.btn_ok}
          titleStyle={style.btn_ok_text}
          title={MODAL_CONSTS.delete_product.answer[1]}
          onPress={() => {
            actionCancel()
            setModalVisible(!modalVisible)
          }}
        />
      </Wrapper>
    </>
  )
}

export default ModalDeletePruduct