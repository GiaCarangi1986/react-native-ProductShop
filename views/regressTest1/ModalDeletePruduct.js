import React from "react";
import { Text, Button, View } from "react-native";
import { MODAL_CONSTS } from "../../const";
import style from "./style";

export const ModalDeletePruduct = ({ setModalVisible = () => { }, modalVisible = false, deleteProducts = () => { }, actionCancel = () => { } }) => {
  return (
    <>
      <Text style={style.text}>{MODAL_CONSTS.delete_product.title}</Text>
      <View>
        <Button
          id='delete'
          type='outline'
          buttonStyle={style.btn_delete}
          titleStyle={style.btn_delete_text}
          title={MODAL_CONSTS.delete_product.answer[0]}
          onPress={() => {
            deleteProducts()
            setModalVisible(!modalVisible)
          }}
        />
        <Button
          id='cancel'
          type='outline'
          buttonStyle={style.btn_ok}
          titleStyle={style.btn_ok_text}
          title={MODAL_CONSTS.delete_product.answer[1]}
          onPress={() => {
            actionCancel()
            setModalVisible(!modalVisible)
          }}
        />
      </View>
    </>
  )
}