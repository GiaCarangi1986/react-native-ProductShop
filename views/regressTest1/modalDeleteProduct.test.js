import 'jsdom-global/register';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { ModalDeletePruduct } from './ModalDeletePruduct';
import { mount, configure } from 'enzyme';

configure({ adapter: new Adapter() });
test('регрессионный тест подтверждения удаления продукта из корзины', () => {
  const mockSetModalVisible = jest.fn()
  const modalVisible = false
  const mockDeleteProducts = jest.fn()
  const mockActionCancel = jest.fn()

  const component = mount(
    <ModalDeletePruduct
      setModalVisible={mockSetModalVisible}
      modalVisible={modalVisible}
      deleteProducts={mockDeleteProducts}
      actionCancel={mockActionCancel}
    />
  )

  expect(component).toMatchSnapshot();

  component.find('#delete').simulate('click')

  expect(mockSetModalVisible).toBeCalledTimes(1)
  expect(mockDeleteProducts).toBeCalledTimes(1)
  expect(mockActionCancel).toBeCalledTimes(0)
});