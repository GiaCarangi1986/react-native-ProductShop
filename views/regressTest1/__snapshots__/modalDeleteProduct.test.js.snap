// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`регрессионный тест подтверждения удаления продукта из корзины 1`] = `ReactWrapper {}`;
